scipy/stats/_stats_py.py

```diff
@@ -4657,11 +4648,18 @@ def statistic(x, y, axis):
     r = xp.asarray(xp.clip(r, -one, one))
     r[const_xy] = xp.nan
 
-    # As explained in the docstring, the distribution of `r` under the null
-    # hypothesis is the beta distribution on (-1, 1) with a = b = n/2 - 1.
-    ab = xp.asarray(n/2 - 1)
-    dist = _SimpleBeta(ab, ab, loc=-1, scale=2)
-    pvalue = _get_pvalue(r, dist, alternative, xp=xp)
+    # Make sure we return exact 1.0 or -1.0 values for n == 2 case as promissed
+    # in the docs.
+    if n == 2:
+        r = xp_sign(r)
+        pvalue = xp.ones_like(r)
```

Comment:
Can you create `pvalue` in one call to [`xp.where`](https://data-apis.org/array-api/latest/API_specification/generated/array_api.where.html)? That will fix the test failure (due to JAX arrays being immutable).